### ISSUE TITLE
Path bug in GitUtil (for use-gitignore flag)

### ISSUE THREAD

**Issue by ocaraveo** [dnGrep/dnGrep#457]
Bug in GitUtil; quotation characters are not trimmed in 'git status' output.

When using "Use .gitignore" option, if command `git status --short --ignored` produces paths with spaces then the resulting lines will have quotation marks, causing an ArgumentException calling [`Path.Combine`](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=netframework-4.6.2) (at [GitUtil.cs:138](https://github.com/dnGrep/dnGrep/blob/master/dnGREP.Common/GitUtil.cs#L138)).

> **System.ArgumentException**: 'Illegal characters: ["] in path.\r\nParameter name: path'

Generally, outside of ArgumentException, the logic in [Gitignore](https://github.com/dnGrep/dnGrep/blob/master/dnGREP.Common/GitUtil.cs#L127,L143) is **not** expecting `list` to have item values that are wrapped in quotations. For example, the logic to determine if a path is a directory or a file will misbehave if path ends with quotation character.

Suggest patching [dnGREP.Common/GitUtil.cs#L82](https://github.com/dnGrep/dnGrep/pull/385/files#diff-7de66cdf34a4c3df1eb967ff6fdd11e1abf8c749719545233cb1944a8f6ddc40R82):
```cs
list.Add(line.Substring(3).Trim('"'));
```

**Comment by doug24** [comment]
Thanks for the report and the analysis.  I will put this fix in, but I haven't been able to reproduce the problem.  I created a test repo with spaces in both paths and file names, but I don't get quotes in the !! lines, just in the M line.  I'm wondering what are you doing differently, or is it a git setting?

```
$ git status --short --ignored
 M .gitignore
 M "WpfApp1 with spaces/WpfApp1 with spaces.csproj"
?? WpfApp1 with spaces/Test Data/
!! .vs/
!! WpfApp1 with spaces/Test Data/TextFile number1.txt
!! WpfApp1 with spaces/bin/
!! WpfApp1 with spaces/obj/

```

**Comment by ocaraveo** [comment]
Without reverting my git client, I can't be sure if this is change in latest version of git, 2.29.2.

Trying to follow your example, here's my sample showing the git command output with mixed quotations output:
![image](https://user-images.githubusercontent.com/6856069/99157319-9571de80-267c-11eb-9c47-fded8265d19b.png)

**Comment by doug24** [comment]
Yes, it was the git version - I was running an older version and when I updated to 2.29.2, I got the error.

Thanks for your help!

**Comment by doug24** [comment]
fixed in 2.9.282
